src/decorators/gb.ts

```diff
@@ -0,0 +1,54 @@
+import IDecorator from "./decorator";
+
+const sixDigitAreaCodes = [
+  "013873",
+  "015242",
+  "015394",
+  "015395",
+  "015396",
+  "016973",
+  "016974",
+  "016977",
+  "016977",
+  "017683",
+  "017684",
+  "017687",
+  "019467",
+];
+
+const decorator: IDecorator = {
+  decorate(phoneNumber: string) {
+    let localNumber;
+
+    if (phoneNumber.substring(2, 4) === "02") {
+      localNumber = `${phoneNumber.substring(3, 5)}` +
+        `${phoneNumber.length > 3 ? ` ${phoneNumber.substring(5, 9)}` : ""}` +
+        `${phoneNumber.length > 5 ? ` ${phoneNumber.substring(9)}` : ""}`;
+    } else if (
+      phoneNumber.substring(2, 4) === "03" ||
+      phoneNumber.substring(2, 4) === "08" ||
+      phoneNumber.substring(2, 4) === "09" ||
+      phoneNumber.substring(2, 4) === "08" ||
+      phoneNumber.substring(2, 5) === "011" ||
+      (phoneNumber.substring(2, 4) === "01" && phoneNumber.charAt(5) === "1")

```

Comment:
Are these files hand written? It seems a bit ad hoc to write these long conditional blocks for specific countries. It's good as a starting point though.

Reply:
I'm not sure there will be another way around it.

Each country has different validation and formatting rules for their phone numbers, And a lot of the rules are antiquated or without much logical sense to them (though I have a feeling that the UK one will be the worst).